crates/sui-cluster-test/src/test_case/coin_merge_split_test.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{helper::verify_gas_coin, TestCaseImpl, TestContext};
+use anyhow::{anyhow, bail};
+use async_trait::async_trait;
+use sui_types::base_types::{ObjectID, SuiAddress};
+use sui_types::object::Owner;
+use tracing::{debug, info};
+
+pub struct CoinMergeSplitTest;
+
+#[async_trait]
+impl TestCaseImpl for CoinMergeSplitTest {
+    fn name(&self) -> &'static str {
+        "CoinMergeSplit"
+    }
+
+    fn description(&self) -> &'static str {
+        "Test merge and split SUI coins"
+    }
+
+    async fn run(&self, ctx: &mut TestContext) -> Result<(), anyhow::Error> {
+        let mut sui_objs = ctx.get_sui_from_faucet(Some(2)).await;
+        let gas_obj = sui_objs.swap_remove(0);
+
+        let signer = ctx.get_wallet_address();
+        let primary_coin = sui_objs.swap_remove(0);
+        let primary_coin_id = *primary_coin.id();
+        let original_value = primary_coin.value();
+
+        // Split
+        info!("Testing coin split.");
+        let amounts = vec![1, (original_value - 2) / 2];
+
+        let data = ctx
+            .get_gateway()
+            .split_coin(
+                signer,
+                *primary_coin.id(),
+                amounts,
+                Some(*gas_obj.id()),
+                5000,
+            )
+            .await
+            .or_else(|e| bail!("Failed to get transaction data for coin split: {}", e))?;
+
+        let split_response = ctx
+            .sign_and_execute(data, "coin split")
+            .await
+            .to_split_coin_response()
+            .or_else(|e| bail!("Failed to execute SplitCoin: {e}"))?;
+
+        // Verify fullnode observes the txn
+        // Let fullnode sync
+        ctx.let_fullnode_sync().await;
+
+        let verification_results = futures::future::join_all(
+            split_response
+                .new_coins
+                .iter()
+                .map(|coin_info| {
+                    verify_gas_coin(
+                        ctx.get_fullnode(),
+                        coin_info.reference.object_id,
+                        Owner::AddressOwner(signer),
+                        false,
+                        true,
```

Comment:
let's not use bool params when we can help it. Can you make it possible to do something like:

```
verify_object(coin_info.reference.object_id)
    .owned_by(signer)
    .is_sui()
```

I think one of the best things we can do to make our tests better in the future is aggressively factor boilerplate like this - writing tests right now often requires tons of copy/paste.